modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1202,7 +1202,12 @@ void ONNXImporter::parseReduce(LayerParams& layerParams, const opencv_onnx::Node
 
 void ONNXImporter::parseSlice(LayerParams& layerParams, const opencv_onnx::NodeProto& node_proto)
 {
-    MatShape inpShape = outShapes[node_proto.input(0)];
+    MatShape inpShape;
+    if (constBlobs.find(node_proto.input(0)) != constBlobs.end() && outShapes[node_proto.input(0)].empty())
```

Comment:
Why we need to check for empty shapes if the question is to handle constant or variable input?

Reply:
I used to think that `constBlobs.find(node_proto.input(0)) != constBlobs.end()` is enough to catch constant input.

Reply:
> I used to think that `constBlobs.find(node_proto.input(0)) != constBlobs.end()` is enough to catch constant input.

In reality, this that enough. I just added it to make sure that we catch exactly the case I am talking about since I was not sure if **constBlobs** get filled on when there is one input. 

If this is crucial I can remove the other part of condition

Reply:
Yes, let's remove please until we find a proper reproducer